### ISSUE TITLE
[REF] travis: Reduce time for job to avoid exceeded the maximum time limit

### ISSUE THREAD

**Issue by moylop260** [OCA/OCB#639]
Currently we have a build too long and travis show the error:
 - [Exceeded the maximum time limit for jobs](https://github.com/travis-ci/travis-ci/issues/5881)

It's because OCB is running tests of all odoo modules.

**Comment by pedrobaeza** [comment]
Yes, but we need to run those tests to make sure everything is OK.

**Comment by moylop260** [comment]
Yes, this issue is to fix both problems.
1) Run all tests
2) No error because exceeded the maximum time limit

**Comment by pedrobaeza** [comment]
Any suggestion for improving the time? Or in general? I can't figure out any.

**Comment by StefanRijnhart** [comment]
Can we cut up the testing into separate jobs? For instance, in one job we run tests on core modules such as account,sale,project,purchase,mrp. Then, in another job we *install* these modules with demo data, and then proceed to install *and* test all other modules. It is not efficient, but it might count as a workaround for the job time limit. Or is the time limit applied to all jobs together?

**Comment by pedrobaeza** [comment]
Each job has its time limit, but the strange thing is that the time limit only counts if there's no output, so it should be something more I think.

**Comment by StefanRijnhart** [comment]
I read about two different limits here: https://www.koszek.com/blog/2016/07/25/dealing-with-large-jobs-on-travis/

```
- time limit on jobs: which currently is set to 30 minutes.
- 10 minute watchdog: if your job doesn’t output anything in this period, it will be killed.
```
Do we have 10 minutes without output in the OCB jobs?

**Comment by pedrobaeza** [comment]
We shouldn't, as there's no large enough jobs with that problem. That limit can be augmented any way to double (20 min) as I did in https://github.com/OCA/l10n-spain/blob/9363c6d767ef0afc2c37a24163e67922c7304032/.travis.yml#L36

**Comment by StefanRijnhart** [comment]
I suffered a timeout on my most recent PR on OCB/8.0. I got it green just now by doing what I set out to do in my comment above: by splitting up the modules in groups (3 of them):

* core modules plus dependencies (sale, account, stock, purchase, project, mrp)
* additional feature modules
* localization modules.

See this build: https://travis-ci.org/OCA/OCB/builds/286650494

To try it out, I hardcoded the modules but this is easily implemented in the MQT travis scripts. I can do it if you like.

**Comment by StefanRijnhart** [comment]
I'm picking this up again. See https://github.com/OCA/maintainer-quality-tools/pull/535 and #722

**Comment by sbidoul** [comment]
Out of curiousity, why can't we use the OCA runbot to test OCB?

**Comment by pedrobaeza** [comment]
Well, the same as for the rest of the repositories: to save own CPU time and to get in parallel both things ASAP: the instance and the tests.

**Comment by pedrobaeza** [comment]
Already done!
